api/src/utils/nova/searchExecutor.ts

```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
+                    this.apiService
+                        .getSlotCommitment(searchQuery.slotIndex)
+                        .then((response) => {
+                            if (searchQuery.slotIndex) {
+                                promisesResult = {
+                                    slotIndex: String(searchQuery.slotIndex),
+                                };
+                                resolve();
+                            } else {
+                                reject(new Error("Slot commitment fetch failed"));
+                            }
+                        })
+                        .catch((_) => {
+                            reject(new Error("Slot commitment fetch failed"));
+                        });
+                }),
             );
         }
 
         if (searchQuery.slotCommitmentId) {
             promises.push(
-                this.executeQuery(
-                    this.apiService.getCommitment(searchQuery.slotCommitmentId),
-                    (result) => {
-                        promisesResult = {
-                            slotIndex: String(result.slot.slot),
-                        };
-                    },
-                    "Slot commitment fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .getCommitment(searchQuery.slotCommitmentId)
+                        .then((response) => {
+                            if (response?.slot?.slot) {
+                                promisesResult = {
+                                    slotIndex: String(response.slot.slot),
+                                };
+                                resolve();
+                            } else {
+                                reject(new Error("Slot commitment fetch failed"));
+                            }
+                        })
+                        .catch((_) => {
+                            reject(new Error("Slot commitment fetch failed"));
+                        });
+                }),
             );
         }
 
         if (searchQuery.tag) {
             promises.push(
-                this.executeQuery(
-                    this.apiService.taggedOutputs(searchQuery.tag),
-                    (response) => {
-                        if (!response.basicOutputs.error || !response.nftOutputs.error) {
-                            promisesResult = {
-                                taggedOutputs: response,
-                            };
-                        }
-                    },
-                    "Tagged details fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .taggedOutputs(searchQuery.tag)
+                        .then((response) => {
+                            if (!response.basicOutputs.error || !response.nftOutputs.error) {
```

Comment:
I don't trust out "errors" too much.
Consider just accepting `if (response?.basicOutputs?.outputs || response?.nftOutputs?.outputs)` to be a strong enough condition to resolve the promise.